starlite/response/base.py

```diff
@@ -130,10 +136,25 @@ def set_cookie(
         httponly: bool = False,
         samesite: Literal["lax", "strict", "none"] = "lax",
     ) -> None:
-        """Ses a cookie on the response.
+        ...
+
+    def set_cookie(  # type: ignore[misc]
+        self,
+        key: Union[str, Cookie],
```

Comment:
I'm not too happy about having to name this one `key`, but in regards of not breaking existing code it's the best solution. Most IDEs will pick up on the `cookie` positional-only arg in the first overload as well, so it's  fine IMO.